### PR TITLE
Improve synth tones example

### PR DESCRIPTION
**Pull request by jamierpond**
Now also plays through a series of waveforms :)

Also here making some aesthetic judgements on naming/convention which am very happy to have a conversation about!

### CODE REVIEW

**Review by est31 — APPROVED** [review]
Looks good, although could you put a trailing 0 instead of trailing dot for the floats?

**Comment by jamierpond** [comment]
@est31 do you know why that windows test might be failing? I imagine that's something that's regressed outside the scope of this PR?

File: examples/synth_tones.rs [hunk 1]
```diff
@@ -82,29 +128,44 @@ pub fn host_device_setup(
...
+
     let stream = device.build_output_stream(
         config,
         move |output: &mut [T], _: &cpal::OutputCallbackInfo| {
-            on_window(output, &mut request, on_sample)
+            // for 0-1s play sine, 1-2s play square, 2-3s play saw, 3-4s play triangle_wave
+            let time_since_start = std::time::Instant::now()
+                .duration_since(time_at_start)
+                .as_secs_f32();
+            if time_since_start < 1. {
```
Comment (est31) [inline examples/synth_tones.rs]: here too

Comment (jamierpond) [reply examples/synth_tones.rs]: oh thanks!

**Review by est31 — APPROVED** [review]

**Comment by hugovdm** [comment]
I have questions about this PR... and at the very least, a fix that makes it play 440Hz instead of the 880Hz it is currently playing:
https://github.com/RustAudio/cpal/pull/792

**Pull request merged**